src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -955,7 +973,11 @@ private SafeProcessHandle GetProcessHandle(int access, bool throwIfExited)
                         }
                     }
                 }
-                return _processHandle;
+
+                // If we dispose of our contained handle we'll be in a bad state. NetFX dealt with this
+                // by doing a try..finally around every usage of GetProcessHandle and only disposed if
```

Comment:
I'm not clear in my head how netfx didn't have an issue here. I see it, like us, checks in `ReleaseProcessHandle` that it's not our own process handle. In what situation would this previously return a handle that would get through that check?